### PR TITLE
 #162233812  admin able send email notifications 

### PR DESCRIPTION
**Pull request by espoirMur**
#### What does this PR do?
- Implement send email notifications feature
#### Description of Task to be completed?
- Implement send email notification feature
- Fix typos in error message and function names and file names
- Create the main error handler for all routes
- Put the database configuration object in config.js
- Improve tests for signin endpoint
- Improve tests for post order endpoint
- Use array constructor when running queries
#### Any background context you want to provide?
- Feedback got on Monday and Tuesday
#### What are the relevant pivotal tracker stories?
 #162233812 
#### Questions:
PS: I'm using a fake SMTP server with nodemailer the mail are not delivered ethereal.com 

But once I found a better server it can be delivered
- *Error handler failed to work in production without next it keeps failing I don't know why*

### CODE REVIEW

File: app/server.js [hunk 1]
```diff
@@ -9,51 +15,51 @@ import {
   joiErrors,
 } from './middlewares/errors';
 import jsonReplacer from './utils/jsonReplacer';
-import { ensureAuthentificated } from './middlewares/authentification';
-import dotenv from 'dotenv';
+import { ensureAuthenticated } from './middlewares/authentification';
```
Comment (andela-ookwuolisa) [inline app/server.js]: change the file name to `authentication`

File: config.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import dotenv from 'dotenv';
+import url from 'url';
+
+dotenv.config();
+
+// read the virtual environement
+const env = process.env.NODE_ENV;
+let dbConfigObject;
+
+if (env === 'HEROKU') {
```
Comment (andela-ookwuolisa) [inline config.js]: change to `PRODUCTION`

File: app/server.js (hunk 1 above)
Comment (espoirMur) [reply app/server.js]: Done

File: config.js (hunk 2 above)
Comment (espoirMur) [reply config.js]: done

**Pull request merged**